### ISSUE TITLE
Timeline then does not work as expected

### ISSUE THREAD

**Issue by bratva** [greensock/GSAP#337]
Hello.

Timeline has method "then" https://greensock.com/docs/v3/GSAP/Timeline/then().
https://jsfiddle.net/eychurwx/1/

```(js)
const tm = gsap.timeline({paused: true});
const elem = document.querySelector('.box');

tm
  .add('start')
  .add(gsap.to(elem, {x: 200}))
  .add('center')
  .add(gsap.to(elem, {x: 400}))
  .add('end');
  
tm.tweenTo('end')
  .then(() => {console.log('1'); return tm.tweenTo('start');})
  .then(() => {console.log('2'); return tm.tweenTo('center');})
  .then(() => {console.log('finish')});
```

I expect to see `1` then `2` then `finish`, but see `1 2` (without delay) and don't see `finish`. 

Is this expected behavior?

**Comment by OSUblake** [comment]
There have been some changes to that method. This fork is using the next version, but it looks like it still might need some work.
https://jsfiddle.net/OSUblake/xoLkgeam/

`finish` is happening, but `1` and `2` are happening at the same time. And the returned value in `2` is undefined, which shouldn't happen.

**Comment by jackdoyle** [comment]
Good catch. I think it's resolved now (get it? "resolved"?)
https://s3-us-west-2.amazonaws.com/s.cdpn.io/16327/gsap-latest-beta.min.js

Better?

**Comment by bratva** [comment]
Yes, thank you Jackdoyle 👍 ! When will it be in npm release?

p.s. I also noticed that Method `reverse()` does not support `then` =(.
https://jsfiddle.net/9c4xf21p/

**Comment by jackdoyle** [comment]
Ah, interesting - I hadn't really thought about then() being triggered onReverseComplete but I suppose that makes sense. Any objections @OSUblake ? 

I've updated the beta - does this work well for you?: https://s3-us-west-2.amazonaws.com/s.cdpn.io/16327/gsap-latest-beta.min.js

As for timing, we're hoping to get it out within the next few days.

**Comment by OSUblake** [comment]
> I hadn't really thought about then() being triggered onReverseComplete 

Me either, but I guess it makes sense.

**Comment by bratva** [comment]
> does this work well for you?
Yes, work. Thank you!

I will wait the release with this fix.

**Comment by jackdoyle** [comment]
Should be resolved in the latest release (3.0.2)
